src/components/Header/Header.jsx

```diff
@@ -1,29 +1,49 @@
-import './header.css';
-import { Link, useLocation } from 'react-router-dom';
-import logo from '../../assets/logo.png';
-import logoutIcon from '../../assets/logout-icon.svg';
+import "./header.css";
+import { Link, useLocation } from "react-router-dom";
+import { useEffect, useState } from "react";
+import logo from "../../assets/logo.png";
+import logoutIcon from "../../assets/logout-icon.svg";
 
 export default function Header() {
   const location = useLocation();
+  const [username, setUsername] = useState(
+    () => localStorage.getItem("username") || ""
+  );
+
+  useEffect(() => {
+    setUsername(localStorage.getItem("username"));
+  }, [localStorage.getItem("username")]);
```

Comment:
This array is called the dependency array, in essence it monitors a variable or data for changes. The invocation of `localStorage.getItem("username")` will return a primitive value that will stay static (for example, a string of "Emily"), it wont actually refetch this each time to check if its a new username. 

You should instead have this listen for the `username` state you created on line 9. This will have the effect of storing the username in localStorage correctly. Great idea though!